crates/context/README.md

```diff
@@ -0,0 +1,65 @@
+
+## Context State
+
+The state of a context is a key-value store, where the key is a 32-byte identifier and the value is a byte array. The state is synchronized across all members of the context, and all state mutations are broadcasted to all members.
+
+## State Mutations
+
+A state mutation is an operation that changes the state of the context. As of the time of writing, state mutations are represented as transactions that define the application method to be called, and it's input (arguments). We make no assumptions about the application, nor it's inputs and outputs, as long as it's clients agree on the schema & serialization method.
+
+All members of the context, following the reception of a transaction, will execute the transaction on their local copy of the state to advance it to the new state.
+
+The unfortunate downside of this approach is that we're unnecessarily executing the same transaction on all nodes without using that information to achieve trustless consensus. We're currently working on a solution to this problem, which reimagines consensus.
+
+## Context Membership
+
+The membership of a context is a list of member identifiers. Each member is identified by a 32-byte key, which represents the public key of the member.
+
+### Invitations
+
+Bob wants to join a context. He shares his public key with Alice, who is already a member of the context. Alice then broadcasts an invitation to all members of the context, which is made up of Bob's public key. Following this invitation, Alice shares the context ID with Bob. At this point, Bob can make a catchup request, which will be responded to by any member of the context who has some of the state and now knows about Bob. Eventually, Bob receives all transactions required to make up the complete state, at which point he is considered a full member of the context. And can proceed to make mutations.
+
+### Leaving
+
+Bob wants to leave a context. He broadcasts a leave request to all members of the context, which is made up of his public key. Following this request, Bob is removed from the membership list, and all members will stop broadcasting state mutations to Bob.
+
+## Encryption
+
+As of the time of writing, all messages are sent in plaintext. But we're currently working on encrypting all messages using distinct, yet, deterministic keys by employing the double-ratchet algorithm, which provides forward secrecy, and post-compromise security. We'll revise this document once this, among other features, is implemented.
+
+## Application Upgrade
+
+As noted [above](#administrative-operations), application authors can define gated behaviour for administrative operations. This includes the ability to upgrade the application. Subsequent transactions made by members from the old version of the application will be rejected by the new version.
+
+## Context Deletion
+
+There's no way to "delete a context" for all members outside of the scope of all members leaving it.
```

Comment:
```
    pub async fn delete_context(
        &self,
        context_id: &calimero_primitives::context::ContextId,
    ) -> eyre::Result<bool> {
        let mut handle = self.store.handle();

        let key = calimero_store::key::ContextMeta::new(*context_id);

        if !handle.has(&key)? {
            return Ok(false);
        }

        handle.delete(&key)?;

        self.unsubscribe(context_id).await?;

        Ok(true)
    }
```

So this does not delete context?

Reply:
That removes the context from your local node, it does not "delete" it from existence

Reply:
It is then more of `leave_context`. Delete is misleading as I as an owner could assume that it does not exists anymore

Reply:
The issue is thinking anything wrt the node happens globally, when we're starting from the premise of local-first.

There's no such thing as global actions / global state, everything is local and node-oriented.

You can certainly create a context that has already been created before.

Reply:
But shouldn't delete context allow me as an owner to kick everyone from the context and then close it?  Others can then create new ones if they want.

We can continue this discussion outside this PR as it is change of concept and not related to doc itself.